### ISSUE TITLE
reply not working when envelope-from / return-path is sent by client

### ISSUE THREAD

**Issue by eleith** [anonaddy/anonaddy#245]
in the [documentation](https://anonaddy.com/self-hosting/), anonaddy has postfix pass emails to the application as such

```
anonaddy unix - n n - - pipe
  flags=F user=johndoe argv=php /var/www/anonaddy/artisan anonaddy:receive-email --sender=${sender} --recipient=${recipient} --local_part=${user} --extension=${extension} --domain=${domain} --size=${size}
```

anonaddy's reply logic in [ReceiveEmail.php](https://github.com/anonaddy/anonaddy/blob/master/app/Console/Commands/ReceiveEmail.php#L144) depends on `${sender}` 

however, according to [postfix's pipe documentation](http://www.postfix.org/pipe.8.html), `${sender}` is derived from the sender envelope from (as opposed to directly using the `From` header).

mail clients / services that want to track bounces set the `Return-Path` header with an alternate email address. This email address is also referred to as the envelope-from address. Thus, when set, the `${sender}` will refer to this alternate email address and not the `From` address.

as a result, this bounce email address will not match any valid recipients and thus all replies will fail to go through.

this issue can be worked around, if after [parsing the raw email](https://github.com/anonaddy/anonaddy/blob/master/app/Console/Commands/ReceiveEmail.php#L71), you extract the sender using the following method

```php
    protected function getSenderFrom()                 
    {                                                            
        $from = $this->parser->getHeader('From');     
                                                                        
        try {     
          $senderFromParsed = mailparse_rfc822_parse_addresses($from);   
          return $senderFromParsed[0]['address'];    
        } catch(\Exception $e) {  
          return $this->option('sender');                                             
        }                                
    } 
```
and pass that value into the various recipient verification calls used to validate and then eventually send the reply.

**Comment by eleith** [comment]
i would prefer a way to do this in postfix directly, but haven't found a way to do this yet.

**Comment by willbrowningme** [comment]
This has been fixed in the latest release [v0.9.0](https://github.com/anonaddy/anonaddy/releases/tag/v0.9.0).
